File: client-data/src/main/java/com/jraska/github/client/UriHandlerViewModel.java
package com.jraska.github.client;

import android.arch.lifecycle.ViewModel;
import okhttp3.HttpUrl;

public final class UriHandlerViewModel extends ViewModel {
  private final DeepLinkHandler deepLinkHandler;

  UriHandlerViewModel(DeepLinkHandler deepLinkHandler) {
    this.deepLinkHandler = deepLinkHandler;
  }

  public void handleDeepLink(String deepLinkText) {
    HttpUrl deepLink = HttpUrl.parse(deepLinkText);
    deepLinkHandler.handleDeepLink(deepLink);
  }
}

File: client-data/src/main/java/com/jraska/github/client/UriHandlerViewModel.kt
package com.jraska.github.client

import androidx.lifecycle.ViewModel
import okhttp3.HttpUrl

class UriHandlerViewModel constructor(private val deepLinkHandler: DeepLinkHandler) : ViewModel() {
  fun handleDeepLink(deepLink: HttpUrl) {
    deepLinkHandler.handleDeepLink(deepLink)
  }
}
